controllers/controllers/shared/index.go

```diff
@@ -3,15 +3,30 @@ package shared
 import (
 	"context"
 
+	servicesv1alpha1 "code.cloudfoundry.org/cf-k8s-controllers/controllers/apis/services/v1alpha1"
+
 	networkingv1alpha1 "code.cloudfoundry.org/cf-k8s-controllers/controllers/apis/networking/v1alpha1"
 
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
-const DestinationAppName = "destinationAppName"
+const (
+	DestinationAppName         = "destinationAppName"
+	ServiceBindingAppGUIDIndex = "serviceBindingAppGUID"
+)
 
 func SetupIndexWithManager(mgr manager.Manager) error {
+	if err := setupRouteIndex(mgr); err != nil {
+		return err
+	}
+	if err := setupServiceBindingIndex(mgr); err != nil {
+		return err
+	}
+	return nil
```

Comment:
This pattern works and is clear, so there's no need to change it. That said, an alternative would be to inline the `IndexField` call here, but move its function argument into its own dedicated function. Something like...

```go
err := mgr.GetFieldIndexer().IndexField(context.Background(), &networkingv1alpha1.CFRoute{}, DestinationAppName, RouteDestinationAppNameIndexFn)
if err != nil {
  return err
} 
```
```go
func RouteDestinationAppNameIndexFn(rawObj client.Object) []string {
			route := rawObj.(*networkingv1alpha1.CFRoute)
			var destinationAppNames []string
			for _, destination := range route.Spec.Destinations {
				destinationAppNames = append(destinationAppNames, destination.AppRef.Name)
			}
			return destinationAppNames
		});
```

Reply:
Good idea, didn't notice that those were actually inline functions. Your suggestion makes a ton of sense!